### PR TITLE
Camera Ready Version for Bioinformatics

### PR DESCRIPTION
**Pull request by millanp95**
This PR makes five key updates:

1. Adds support for direct download of the dataset from HF.

2. **Fix & refactor CNN baseline:** I made sure to correctly load the CNN model (as previously discussed). I also eliminated repeated boilerplate by refactoring shared logic. The results are logegd to W&B.

3. **Upgrade HyenaDNA baseline:**  Replace the out-of-date, locally downloaded model with the official Hugging Face release. Remove Git-LFS dependency. The updated scores differ slightly from the old model, but remain consistent with BarcodeMamba, and BarcodeBERT still leads.

4. **Switch UMAP distance to cosine distance:** Align UMAP’s metric with our KNN similarity searches.      Improves overall embeddings quality for clustering; the relative ordering of models remains unchanged (DNABERT still outperforms and BarcodeBERT is second).

5. **General cleanup:** I removed redundant comments and debug print statements.I also tried to enhance remaining comments for clarity and context.

### CODE REVIEW

File: baselines/linear_probing.py [hunk 1]
```diff
@@ -150,7 +147,7 @@ def run(config):
     print("Labels shapes", y.shape, y_val.shape, y_test.shape)
 
     train = torch.utils.data.TensorDataset(X, y)
-    train_loader = DataLoader(train, batch_size=64, shuffle=True)
+    train_loader = DataLoader(train, batch_size=1024, shuffle=True)
```
[the remark this answers is not in the document]
Comment (millanp95) [reply baselines/linear_probing.py:150]: Here we are training just a single layer, so this should be ok.

File: baselines/zsc.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+
+def zsc_pipeline(X, y_true, metric="cosine", n_neighbours=10):
+
+    # Step 1: Dimensionality reduction with UMAP to 50 dimensions
+    umap_reducer = umap.UMAP(n_components=50, random_state=42, metric=metric, n_neighbors=n_neighbours)
+    X_reduced = umap_reducer.fit_transform(X)
+
+    # Step 2: Cluster the reduced embeddings with Agglomerative Clustering (L2, Ward’s method)
+    agglomerative_clustering = AgglomerativeClustering(n_clusters=3479, linkage="ward")  # 3479 species 3950 bins
```
[the remark this answers is not in the document]
Comment (millanp95) [reply baselines/zsc.py:25]: I though about this but `y_true` contains values in `bin_uri` and in the ZSC pipeline we use the number of species. I think this should be fine too, but I can add it as kwarg of the function

File: baselines/io.py [hunk 3]
```diff
@@ -32,19 +35,16 @@ def load_baseline_model(backbone_name, *args, **kwargs):
...
         "DNABERT-S": (["zhihan1996/DNABERT-S"], kwargs),
-        "DNABERT": (["/scratch/ssd004/scratch/pmillana/checkpoints/dnabert/6-new-12w-0"], kwargs),
+        "DNABERT": (["pretrained_models/6-new-12w-0"], kwargs),
         "BarcodeBERT": ([], kwargs),
     }
 
     out_dimensions = {
         "NT": 512,
-        "Hyena_DNA": 128,
+        "Hyena_DNA": 256,
```
[the remark this answers is not in the document]
Comment (millanp95) [reply baselines/io.py:47]: A couple of reasons, it was cumbersome to manually download the old architecture from the old version of the repo using git-lfs, copying the model to a desired folder and passing that path as an argument to the baseline script. In this way we can cleanly load from FH. Also, this model should be superior to the other one and was used in BarcodeMamba, I though that would strengthen our results.

File: CC_requirements.txt [hunk 4]
```diff
@@ -14,6 +14,7 @@ scikit-learn==1.3+computecanada
 scipy==1.12+computecanada
 seaborn
 torch==2.1.1+computecanada
+torchdata==0.7.1+computecanada
```
[the remark this answers is not in the document]
Comment (millanp95) [reply CC_requirements.txt:17]: Oh yes, torchdata is not used anymore

**Review by Niousha12 — APPROVED** [review]

File: baselines/io.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (millanp95) [reply baselines/io.py:47]: I completed it over the weekend; the values have been updated accordingly in the tables. See `diff.tex` in Overleaf

**Review by scottclowe — APPROVED** [review]
